### DISCUSSION TITLE
Multiple Authentication Questions

### DISCUSSION THREAD

**Discussion by steffen-stolze** [supabase/supabase#7597] [General]
Hey there,

I'm developing a SvelteKit application and want to use Supabase as BaaS. I need some help understanding some things :)
It would be very helpful if someone could give some feedback on what I got right or wrong.

1.
As soon as you create a Supabase client somewhere in the frontend, it initialises the underlying GoTrueClient which will observe route changes. If you have an empty localStorage and click on a password reset link like this: 

`https://xxxx.supabase.co/auth/v1/verify?token=lfhazbfrjbzftpkiwdoa&type=recovery&redirect_to=http://localhost:3000`

the browser will redirect (according to the location response header) to something like this:

`http://localhost:3000/#access_token=eyJhbGci...6u8Qg&expires_in=3600&refresh_token=o3MrO...Rl0Q&token_type=bearer&type=recovery`

and also set two cookies `sb-access-token` and `sb-refresh-token`.

At this point, Supabase will grab the tokens from the URL fragments (or cookies?) and create a `supabase.auth.token` object in the localStorage, containing the user and the access_token. It will also remove the cookies.

If one does not want to have an object containing the access token in the localStorage, the only way would be, **not** to have a Supabase client in the frontend at all, because at some point users will reach the application via a PW reset link with an access_token in the URL?

2.
Initiating a password reset with Supabase will send an email that has a link included (like the one above).

Following the link and having a Supabase client initiated in the frontend will basically log the user in.

At this points it's pretty much the same as a magic link, besides one difference: 
the `supabase.auth.onAuthStateChange` function will fire a **PASSWORD_RECOVERY** in one case and. **SIGNED_IN** in the other?

3.
If one would not want the Supabase token in my localStorage I would have to be careful not to create a Supabase client in the frontend part and not to import the supabase-js client?

If I do this I'd have to deal with password reset URLs containing tokens myself, right?

Is this the only downside If I otherwise would be fine with dealing with Supabase only from the backend endpoints?

4.
I want to make requests from the backend endpoints. Thus I need the access_token on the server side.
I do this by logging in on the server side with a login post endpoint request and setting two cookies in the response headers `sb-access-token` and `sb-refresh-token`.
In subsequent requests to the server side endpoints it seems to be enough to extract this token from the cookie and authenticate with the access_token in the Supabase client:

```
//../todos/index.json.js

import cookie from 'cookie'
import { supabase } from '../../lib/supabase';
export async function get({ request }) {
    const cookies = cookie.parse(request.headers.get('cookie'))
    const user = await supabase.auth.api.getUser(cookies['sb-access-token'])
    await supabase.auth.setAuth(cookies['sb-access-token'])
    const { data, error } = await supabase.from('todos').select()

    if (error) {
        console.error(error)
    }

    return {
        body: {
            todos: data
        }
    }
}
```

5.
For convenience I'd use both, client and server side Supabase code. Most convenience would be the `supabase.auth.onAuthStateChange` function in the browser and being able to directly query the DB from the browser if I need to.
Server side would be useful if I'd need to interact with other systems and APIs that I don't want to expose to the browser.

Is there anything speaking against this approach (besides having a token in localStorage...)?

Thanks and BR

**Comment by shinriyo** [comment]
I also did the tutorial
https://supabase.com/docs/guides/getting-started/tutorials/with-svelte

But I can't go `http://localhost:5173/`

https://MY_URL.supabase.co/auth/v1/verify?token=BLABLABLA&type=signup&redirect_to=http://localhost:5173/

**Reply by shinriyo** [reply]
Is It bug?
